compiler/noirc_evaluator/src/ssa/opt/simple_optimization.rs

```diff
@@ -0,0 +1,103 @@
+use crate::ssa::ir::{
+    basic_block::BasicBlockId,
+    dfg::DataFlowGraph,
+    function::Function,
+    instruction::{Instruction, InstructionId},
+    value::{ValueId, ValueMapping},
+};
+
+impl Function {
+    /// Performs a simple optimization according to the given callback.
+    ///
+    /// The function's reachable blocks are traversed in turn, and instructions in those blocks
+    /// are then traversed in turn. For each one, `f` will be called with a context.
+    ///
+    /// The current instruction will be inserted at the end of the callback given to `mutate` unless
+    /// `remove_current_instruction` or `insert_current_instruction` are called.
+    ///
+    /// `insert_current_instruction` is useful if you need to insert new instructions after the current
+    /// one, so this can be done before the callback ends.
+    ///
+    /// `replace_value` can be used to replace a value with another one. This substitution will be
+    /// performed in all subsequent instructions.
```

Comment:
Let's also add a safety comment that this function uses `reachable_blocks` internally. Replacing instructions which depend upon the ordering of blocks with this method can potentially lead to correctness errors.

Reply:
We could also rename `simple_optimization` to `simple_reachable_blocks_optimization` to make this clearer. It's a bit longer but it's just one call per SSA pass that uses it. Then eventually we could add one for post order if needed. What do you think?

Reply:
I ended up renaming the method plus adding a link to `reachable_blocks` in the doc comments.

Reply:
Thanks! I like the rename as well.